x-pack/plugins/security/public/session/unauthorized_response_http_interceptor.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { HttpInterceptor, HttpErrorResponse, HttpInterceptController } from 'src/core/public';
+
+import { AnonymousPaths } from '../anonymous_paths';
+import { SessionExpired } from './session_expired';
+
+export class UnauthorizedResponseHttpInterceptor implements HttpInterceptor {
+  constructor(private sessionExpired: SessionExpired, private anonymousPaths: AnonymousPaths) {}
+
+  responseError(httpErrorResponse: HttpErrorResponse, controller: HttpInterceptController) {
+    if (this.anonymousPaths.isAnonymous(window.location.pathname)) {
+      return;
+    }
+
+    // if the request was omitting credentials it's to an anonymous endpoint
+    // (for example to login) and we don't wish to ever redirect
+    // TODO: Why is httpErrorResponse.request potentially undefined?
```

Comment:
@eliperelman with the changes introduced by https://github.com/elastic/kibana/pull/42755, both the `HttpResponse` and `HttpResponseError` have the `response` potentially being undefined. I'm struggling to think of a situation where we'd have a http response or a response error where a request shouldn't be there as well?

Reply:
Remove TODO?